08_cell_line_prediction/nbconverted/lasso_range_all.py

```diff
@@ -0,0 +1,374 @@
+    best_perf_df.cv_aupr - best_perf_df.test_aupr
+)
+
+print(best_perf_df.shape)
+best_perf_df.sort_values(by='cv_test_aupr_diff', ascending=False).head()
+
+
+# In[16]:
+
+
+plot_df = (best_perf_df
+    .drop(columns=['cv_test_auroc_diff', 'cv_test_aupr_diff', 'cv_auroc', 'test_auroc'])
+    .melt(id_vars=['gene', 'top_lasso_param', 'seed', 'fold'],
+          value_vars=['cv_aupr', 'test_aupr'],
+          var_name=['dataset_metric'])
+)
+plot_df.head()
+
+
+# In[17]:
+
+
+# plot cv/test performance distribution for each gene
+sns.set({'figure.figsize': (28, 10)})
+sns.set_style('ticks')
+
+fig, axarr = plt.subplots(2, 1)
+
+# order boxes by cv performance per gene
+cv_gene_order = (plot_df[plot_df.dataset_metric == 'cv_aupr']
+    .groupby(['gene', 'top_lasso_param'])
+    .agg(np.median)
+    .sort_values(by='value', ascending=False)
+).index.get_level_values(0).values
+
+# order boxes by test performance per gene
+test_gene_order = (plot_df[plot_df.dataset_metric == 'test_aupr']
+    .groupby(['gene', 'top_lasso_param'])
+    .agg(np.median)
+    .sort_values(by='value', ascending=False)
+).index.get_level_values(0).values
+
+with sns.plotting_context('notebook', font_scale=1.5):
+    sns.boxplot(data=plot_df, order=cv_gene_order,
+                x='gene', y='value', hue='dataset_metric', ax=axarr[0])
+    axarr[0].axhline(0.0, linestyle='--', color='grey')
+    axarr[0].tick_params(axis='x', rotation=90, labelsize=16)
+    axarr[0].tick_params(axis='y', labelsize=16)
+    axarr[0].set_xlabel('Gene (sorted by validation set performance)', size=18)
+    axarr[0].set_ylabel('AUPR', size=18)
+    
+    sns.boxplot(data=plot_df, order=test_gene_order,
+                x='gene', y='value', hue='dataset_metric', ax=axarr[1])
+    axarr[1].legend([], [], frameon=False)
+    axarr[1].axhline(0.0, linestyle='--', color='grey')
+    axarr[1].tick_params(axis='x', rotation=90, labelsize=16)
+    axarr[1].tick_params(axis='y', labelsize=16)
+    axarr[1].set_xlabel('Gene (sorted by test set performance)', size=18)
+    axarr[1].set_ylabel('AUPR', size=18)
+    
+    plt.suptitle(f'Mutation prediction performance on validation/test sets, by gene', y=1.0025)
+    
+plt.tight_layout()
+
+
+# In[18]:
+
+
+# plot difference in validation and test performance for each gene
+sns.set({'figure.figsize': (28, 6)})
+sns.set_style('ticks')
+
+# order boxes by median (cv - test) diff per gene
+gene_order = (best_perf_df
+    .groupby(['gene', 'top_lasso_param'])
+    .agg(np.median)
+    .sort_values(by='cv_test_aupr_diff', ascending=False)
+).index.get_level_values(0).values
+
+with sns.plotting_context('notebook', font_scale=1.5):
+    ax = sns.boxplot(data=best_perf_df, order=gene_order,
+                     x='gene', y='cv_test_aupr_diff',
+                     palette='flare')
+    ax.axhline(0.0, linestyle='--', color='grey')
+    plt.xticks(rotation=90)
+    plt.title(f'Difference between TCGA and CCLE mutation prediction performance, by gene', y=1.02)
```

Comment:
It would be nice to see both average performance and difference in a single plot, e.g. that TP53 isn't much better or worse on CCLE but that's because it scores well on both vs. a gene with moderate performance on one and dismal performance on another. Maybe a scatterplot with mean validation AUPR on one axis and difference in validation and test AUPR on the other axis? Or would that be too messy?

Reply:
That's a good idea! I was planning to add a "filtering" step where I get rid of bad models (we came up with a way to do this in my last paper that I can just use here again), but I think it would be interesting to look at the scatter plot too.